eZ/Bundle/EzPublishMigrationBundle/Command/LegacyStorage/UpdateSortKeysCommand.php

```diff
@@ -0,0 +1,274 @@
+            ->addArgument( "fieldtype_identifier", InputArgument::REQUIRED, "Field type identifier" )
+            ->addArgument( "bulk_count", InputArgument::OPTIONAL, "Number of Content versions processed at once", 100 )
+            ->addOption( "dry-run", null, InputOption::VALUE_NONE, "Execute a dry run" )
+            ->setHelp(
+                <<<EOT
+The command <info>%command.name%</info> updates field
+sort keys in configured Legacy Storage database for a given field type.
+
+Fields will be processed per Content version and updated only if newly generated
+sort key does not match stored one. In this case all field data will be updated.
+
+<warning>During the script execution the database should not be modified.
+
+To avoid surprises (particularly if using custom field types) you are advised to
+create a backup or execute a dry run before proceeding with actual update.</warning>
+
+Since this script can potentially run for a very long time, to avoid memory
+exhaustion run it in production environment using <info>--env=prod</info> switch.
+
+EOT
+            );
+    }
+
+    protected function execute( InputInterface $input, OutputInterface $output )
+    {
+        $fieldTypeIdentifier = $input->getArgument( 'fieldtype_identifier' );
+        $bulkCount = $input->getArgument( 'bulk_count' );
+        $dryRun = $input->getOption( 'dry-run' );
+
+        /** @var \eZ\Publish\Core\Persistence\Database\DatabaseHandler $databaseHandler */
+        $databaseHandler = $this->getContainer()->get( 'ezpublish.connection' );
+
+        /** @var \eZ\Publish\API\Repository\Repository $repository */
+        $repository = $this->getContainer()->get( "ezpublish.api.repository" );
+        $repository->setCurrentUser(
+            $repository->getUserService()->loadUser( 14 )
+        );
+
+        if ( !$this->getContainer()->has( "ezpublish.fieldtype.{$fieldTypeIdentifier}" ) )
+        {
+            $output->writeln( "<error>Field type '{$fieldTypeIdentifier}' was not found.</error> " );
+            return;
+        }
+
+        $warningStyle = new OutputFormatterStyle( "red" );
+        $output->getFormatter()->setStyle( "warning", $warningStyle );
+        $output->writeln( $this->getProcessedHelp() );
+
+        $query = $databaseHandler->createSelectQuery();
+        $query
+            ->select(
+                $query->expr->count(
+                    "DISTINCT " . $query->expr->concat(
+                        $databaseHandler->quoteColumn( "contentobject_id" ),
+                        $query->bindValue( "-", null, PDO::PARAM_STR ),
+                        $databaseHandler->quoteColumn( "version" )
+                    )
+                )
+            )
+            ->from( "ezcontentobject_attribute" )
+            ->where(
+                $query->expr->eq(
+                    $databaseHandler->quoteColumn( "data_type_string" ),
+                    $query->bindValue( $fieldTypeIdentifier )
```

Comment:
Does this fetch all rows from ezcontentobject_attribute of the given data type string, without limits? If so, could be a problem. Tends to be crowded, that table.